### ISSUE TITLE
imread patch raises FileNotFoundError instead of returning None on Windows

### ISSUE THREAD

**Issue by mariameraz** [ultralytics/ultralytics#24405]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

Hi!

I would like a report an inconsistency I found in the imread patch ([L21](https://github.com/ultralytics/ultralytics/blob/9047b77323f94d4f2797b2b49a07a9b2f44a8362/ultralytics/utils/patches.py#L21)). 

According to the docstring, the patched function should return `None` when reading fails. However, `np.fromfile` return `FileNotFoundError` ([L35](https://github.com/ultralytics/ultralytics/blob/9047b77323f94d4f2797b2b49a07a9b2f44a8362/ultralytics/utils/patches.py#L35)) instead of returning `None` when the file does not exist, breaking the documented behavior. 

In my case, my code expects `cv2.imread` to return `None` when a file doesn't exist. However, after importing Ultralytics and testing on multiple operating systems, this assumption breaks only on Windows since the patch is applied globally ([1486](https://github.com/ultralytics/ultralytics/blob/9047b77323f94d4f2797b2b49a07a9b2f44a8362/ultralytics/utils/__init__.py#L1486)), raising `FileNotFoundError` instead.

I applied this fix on my side to handle it for now:

```python
try:
    file_bytes = np.fromfile(filename, np.uint8)
except (FileNotFoundError, OSError):
    return None
```

Thank you!

### Environment

Ultralytics 8.4.36 🚀 Python-3.13.7 torch-2.11.0 CPU (Apple M1)
Setup complete ✅ (8 CPUs, 16.0 GB RAM, 209.7/228.3 GB disk)

OS                     macOS-15.5-arm64-arm-64bit-Mach-O
Environment            Darwin
Python                 3.13.7
Install                git
Path                   /Users/alejandra/Documents/traitly/traitly_test/lib/python3.13/site-packages/ultralytics
RAM                    16.00 GB
Disk                   209.7/228.3 GB
CPU                    Apple M1
CPU count              8
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.4.4>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 12.2.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.33.1>=2.23.0
scipy                  ✅ 1.17.1>=1.4.1
torch                  ✅ 2.11.0>=1.8.0
torch                  ✅ 2.11.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.26.0>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.39.3>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18
{'OS': 'macOS-15.5-arm64-arm-64bit-Mach-O', 'Environment': 'Darwin', 'Python': '3.13.7', 'Install': 'git', 'Path': '/Users/alejandra/Documents/traitly/traitly_test/lib/python3.13/site-packages/ultralytics', 'RAM': '16.00 GB', 'Disk': '209.7/228.3 GB', 'CPU': 'Apple M1', 'CPU count': 8, 'GPU': None, 'GPU count': None, 'CUDA': None, 'Package Info': {'numpy': '✅ 2.4.4>=1.23.0', 'matplotlib': '✅ 3.10.8>=3.3.0', 'opencv-python': '✅ 4.13.0.92>=4.6.0', 'pillow': '✅ 12.2.0>=7.1.2', 'pyyaml': '✅ 6.0.3>=5.3.1', 'requests': '✅ 2.33.1>=2.23.0', 'scipy': '✅ 1.17.1>=1.4.1', 'torch': '✅ 2.11.0!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.26.0>=0.9.0', 'psutil': '✅ 7.2.2>=5.8.0', 'polars': '✅ 1.39.3>=0.20.0', 'ultralytics-thop': '✅ 2.0.18>=2.0.18'}}

### Minimal Reproducible Example

```yaml
name: Reproduce ultralytics patch issue

on: workflow_dispatch

jobs:
  reproduce:
    runs-on: windows-latest
    steps:
      - name: Set up python
        uses: actions/setup-python@v5
        with:
          python-version: "3.12"

      - name: Install dependencies
        run: pip install ultralytics opencv-python

      - name: Reproduce imread patch issue
        run: |
          python -c "
          import cv2
          from ultralytics import YOLO
          img = cv2.imread('img.jpg')
          print(img)
          "
```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mariameraz, thank you for the clear bug report and for including a reproducible Windows workflow 🚀. This is an automated response to help get the issue triaged quickly, and an Ultralytics engineer will also assist you soon.

We appreciate the report about the `imread` patch behavior on Windows, especially the note about documented behavior versus observed `FileNotFoundError`, along with the environment details and proposed workaround.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In this case, thank you — you already included one ✅

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

We also recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Thanks for reporting

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24406: Return `None` in `imread` on read failure to match `cv2.imread`](https://github.com/ultralytics/ultralytics/pull/24406).

This change may resolve the issue by making `imread()` more robust when a file is missing, invalid, or inaccessible:

- adds a `try/except` around `np.fromfile()` in `ultralytics/utils/patches.py`
- catches `FileNotFoundError` and `OSError`
- returns `None` on read failure instead of raising an exception
- keeps existing TIFF and multi-frame decoding behavior unchanged

Credit to @Y-T-G for the PR, with contributions from @Laughing-q.

If you'd like to test the fix now, you can install the latest `main` branch directly:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Or, if preferred, you can wait for the next official PyPI release.

If possible, please test and let us know whether these changes resolve the issue on your side.

Thanks 🙏 for reporting this. If the issue still persists after testing, any additional details or reproduction info would be very helpful.
